### PR TITLE
common/file_util: Make use of std::filesystem

### PR DESCRIPTION
**Pull request by lioncash**
Migrates a bunch of our file handling code over to std::filesystem to simplify a bunch of our code and make it easier to maintain.

Notably with this, `CopyDir` will now function on Windows.

### CODE REVIEW

File: src/common/file_util.h [hunk 1]
```diff
@@ -38,48 +39,34 @@ enum class UserPath {
...
-    bool isDirectory;
-    u64 size;                 // file length or number of entries from children
-    std::string physicalName; // name on disk
-    std::string virtualName;  // name in FST names table
-    std::vector<FSTEntry> children;
-};
-
-// Returns true if file filename exists
-[[nodiscard]] bool Exists(const std::string& filename);
+// Returns true if the exists
```
Comment (ReinUsesLisp) [inline src/common/file_util.h:42]:
```suggestion
// Returns true if the path exists
```

File: src/common/file_util.cpp [hunk 2]
```diff
@@ -67,290 +68,102 @@
...
-        LOG_DEBUG(Common_Filesystem, "path exists {}", fullPath);
-        return true;
-    }
-
-    std::size_t position = 0;
-    while (true) {
-        // Find next sub path
-        position = fullPath.find(DIR_SEP_CHR, position);
+bool Rename(const fs::path& src, const fs::path& dst) {
+    LOG_TRACE(Common_Filesystem, "{} --> {}", src.string(), dst.string());
```
Comment (ReinUsesLisp) [inline src/common/file_util.cpp:127]: Is the double `-` in `-->` intentional? Ditto for `Copy`.

Comment (lioncash) [reply src/common/file_util.cpp:127]: This is a pre-existing log call

**Review by ReinUsesLisp — APPROVED** [review]

**Review by adityaruplaha — APPROVED** [review]
Edit: Nvm, didn't notice this was already merged.

**Pull request merged**